File: java/cosmicsim/src/main/java/com/ddamiani/cosmicsim/ResultStore.java
package com.ddamiani.cosmicsim;

/**
 * An object for storing the results of a simulation run.
 */
public final class ResultStore {
    public static final double DEFAULT_SAMPLE_POINT = 28.0;
    private final double mSamplePoint;
    private int mNumCharged;
    private int mNumNeutral;

    protected ResultStore() {
        this(DEFAULT_SAMPLE_POINT);
    }

    protected ResultStore(double samplingPoint) {
        mSamplePoint = samplingPoint;
        mNumCharged = 0;
        mNumNeutral = 0;
    }

    protected double getSamplingPoint() {
        return mSamplePoint;
    }

    protected int getTotalNumParticles() {
        return mNumCharged + mNumNeutral;
    }

    protected int getNumChargedParticles() {
        return mNumCharged;
    }

    protected int getNumNeutralParticles() {
        return mNumNeutral;
    }

    protected void incrementNumChargedParticles() {
        mNumCharged += 1;
    }

    protected void incrementNumNeutralParticles() {
        mNumNeutral += 1;
    }
}

File: cpp/src/ResultStore.cpp
#include "ResultStore.h"

ResultStore::ResultStore() :
  m_sample_point(DEFAULT_SAMPLE_POINT),
  m_num_charged(0),
  m_num_neutral(0)
{}

ResultStore::ResultStore(double sampling_point) :
  m_sample_point(sampling_point),
  m_num_charged(0),
  m_num_neutral(0)
{}

ResultStore::~ResultStore() {}
